contracts/GenArt721CoreV3.sol

```diff
@@ -808,34 +808,30 @@ contract GenArt721CoreV3 is
         onlyUnlocked(_projectId)
         onlyArtistOrAdminACL(_projectId, this.removeProjectLastScript.selector)
     {
-        require(
-            projects[_projectId].scriptCount > 0,
-            "there are no scripts to remove"
-        );
-        delete projects[_projectId].scripts[
-            projects[_projectId].scriptCount - 1
-        ];
-        projects[_projectId].scriptCount = projects[_projectId].scriptCount - 1;
+        Project storage project = projects[_projectId];
+        require(project.scriptCount > 0, "there are no scripts to remove");
+        delete project.scripts[project.scriptCount - 1];
+        project.scriptCount = project.scriptCount - 1;
         emit ProjectUpdated(_projectId, FIELD_PROJECT_SCRIPT);
     }
 
     /**
      * @notice Updates script type for project `_projectId`.
      * @param _projectId Project to be updated.
-     * @param _scriptType Libary to be injected by renderer. e.g. "p5js"
-     * @param _scriptTypeVersion Version of library to be injected. e.g. "1.0.0"
+     * @param _scriptTypeAndVersion Script type and version e.g. "p5js@1.0.0",
+     * as bytes32 encoded string.
      */
     function updateProjectScriptType(
         uint256 _projectId,
-        string memory _scriptType,
-        string memory _scriptTypeVersion
+        bytes32 _scriptTypeAndVersion
     )
         external
         onlyUnlocked(_projectId)
         onlyArtistOrAdminACL(_projectId, this.updateProjectScriptType.selector)
     {
-        projects[_projectId].scriptType = _scriptType;
-        projects[_projectId].scriptTypeVersion = _scriptTypeVersion;
+        Project storage project = projects[_projectId];
+        // TODO - require @ symbol in _scriptTypeAndVersion
```

Comment:
This PR generally LGTM but we probably shouldn't merge until this part is implemented as well–WDYT?

Reply:
@jakerockland Yes this one was not quite ready for review yet. The on-chain validation and PR description are updated, and are ready for review now! 😄

Reply:
ohh I think something automatically triggered your review request? 🤔 sorry it was early

Reply:
Ah–I think it is because this wasn't a draft maybe, so auto-reviewer assigned?

Reply:
yeah must have been not a draft, oops!